### ISSUE TITLE
Custom Types have wrong `_type` key

### ISSUE THREAD

**Issue by alexbchr** [ricokahler/sanity-codegen#27]
Let's have the following simple schemas:

```js
const fooImage = {
  type: 'image',
  title: 'Top-level Image Type',
  name: 'foo_image',
  fields: [
    {
      title: 'Title',
      name: 'title',
      type: 'string',
    },
  ],
};

const bar = {
  type: 'document',
  title: 'Bar',
  name: 'bar',
  fields: [
    {
      title: 'Nested Image Type',
      name: 'nestedImage',
      type: 'foo_image',
    }
  ],
};
```

When generated, the `FooImage` type is outputted like so:
```ts
export type FooImage = {
  _type: "image";
  asset: SanityAsset;
  /**
   * Title — `string`
   *
   *
   */
  title?: string;
};
```

However, the `_type` property is wrongly represented, as the `_type` of it should in fact be `foo_image` when referenced from the `Bar` document.

I don't see any use case requiring to have the "base type" present in the `_type` property, but I may be wrong. Having the custom type there would better represent the actual Sanity API response format.

**Comment by ricokahler** [comment]
Oh yeah you're absolutely right. This is a definitely a bug, thanks for pointing it out!

I can get a fix out later tonight but if you wanted to take a crack at it, I'll definitely accept a PR.

> I don't see any use case requiring to have the "base type" present in the _type property, but I may be wrong. Having the custom type there would better represent the actual Sanity API response format.

Agreed. The only change needed is that `_type` should return `foo_image` instead of `image`.

**Comment by alexbchr** [comment]
Great! I took a quick try at it sooner today and it proved to be maybe more complex to do that what I initially thought. Unfortunately, I don't have enough time today to dig more into it. 

Just to add to the example, it is not only `image` which is problematic. This should be the behavior for all Types (`image`, `file`, `slug`, `object` and so on...). I did a very quick fix this morning (only for image and file types), [here is the code](https://github.com/ricokahler/sanity-codegen/compare/main...Trampoline-CX:fix/custom-type) (quite incomplete and a test still, but feel free to use it).

**Comment by ricokahler** [comment]
I merged a fix that should fix it for images and files but I think slug is still an issue. I'll address in a separate PR once I get some time.

**Comment by ricokahler** [comment]
@alexbchr A full fix for this should be out in v0.5.0. Let me know if that does it for you!

**Comment by alexbchr** [comment]
@ricokahler Just tested with latest (0.7.0), and everything seems to work perfectly! Thanks for the quick fix! 🎉
